app/src/main/java/com/udacity/nanodegree/blooddonation/base/PresenterFactory.java

```diff
@@ -0,0 +1,31 @@
+package com.udacity.nanodegree.blooddonation.base;
```

Comment:
This code, doesn't makes sense. Having a factory class for all presenters, serves no purpose.What, if you have more than 20 presenters in the code?

Make sense module wise tough, let's say for login you can have a loginfactory for  different presenters, like google signin or fb sigin and all.

Reply:
Hey Can you explain this @ritesh-singh

Reply:
Oh wait got it . Well sorry to point out but you wrong in this case... If you time to check. In BaseActivity in PR . Line line 30 u will find PresenterFactory.inject(this). Using this will remove the headache of adding preseter intilizing at every class. Not only that the Presenter will enable you to know all your presenters of your entire app. Make you understand where you entire Algo exisits